#158487974 Game-Log API Endpoints

iverenshaguy opened the pull request:
#### What does this PR do?
This pull request updates the game log api endpoints

#### Description of Task to be completed?
- update existing profile page endpoint data
- add get donations api endpoints
- update game history api endpoints
- use neccessary api endpoints in profile controller
- implement view for empty game history, leaderboard and donations
- implement cards for game history
- update tests

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#158487974

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-07-12 at 5 10 12 pm" src="https://user-images.githubusercontent.com/22734663/42645764-c9419cba-85f6-11e8-9996-7c3d6ec63b56.png">


#### Questions:
N/A

iverenshaguy commented:
I did not write any tests or new JS code so I don't understand how the code coverage decreased from my closed PR to this reopened one

ozimos reviewed (approved):
Excellent work,
All tests passing and test coverage increased. Keep it up

Stylll commented on client/views/profile.html:
```diff
@@ -81,63 +81,79 @@
+                  <div class="empty-section no-games-played container" ng-show="gamesPlayed == 0">
+                    Welcome Newbie. <a href="/#!/app">Join a Game</a>
+                  </div>
+                  <div class="games-card-group" ng-show="gamesPlayed != 0">
```
Quick question, is it possible for the variable gamesPlayed to be less that zero ?. If it is, then I think your checks should be "> 0" and not "!= 0".

iverenshaguy replied:
For some reason that doesn't work. Tried it already and I just tried it again. 
<img width="719" alt="screen shot 2018-07-12 at 10 57 27 pm" src="https://user-images.githubusercontent.com/22734663/42661742-fc9df696-8626-11e8-8461-d1cdb75fd1ac.png">
<img width="1241" alt="screen shot 2018-07-12 at 10 57 53 pm" src="https://user-images.githubusercontent.com/22734663/42661758-0ad3eee6-8627-11e8-94ec-4a84632c11e2.png">
<img width="822" alt="screen shot 2018-07-12 at 10 58 29 pm" src="https://user-images.githubusercontent.com/22734663/42661795-29dacdbe-8627-11e8-8fda-c894ce7cc4ab.png">
<img width="938" alt="screen shot 2018-07-12 at 10 58 45 pm" src="https://user-images.githubusercontent.com/22734663/42661800-30ad6174-8627-11e8-8c04-37aaee34eb66.png">

Stylll commented on server/src/controllers/api/users.js:
```diff
@@ -176,21 +169,47 @@ class UsersApiController {
         if (err) return next(err);
 
-        res.json({
+        res.status(201).json({
```
I noticed you're using the "findOneAndUpdate" method which is used to update an entry in the database but your response status code is 201 which is the response code to use when a new entry is created. Is there a reason why you are returning a 201 status code instead of a 200 status code ?.

iverenshaguy replied:
Yes. I did that because it updates the User collection but actually creates a new Donation. The Donations array is part of the User Collection. So I changed it to 201 from 200 because 200 didn't seem right since it's a creation. Tomorrow, during our TTL interaction, the team would make a decision on which status code is the better for this case.

Stylll replied:
Alright. Brilliant decision.

iverenshaguy replied:
We decided to go with 200 and I have updated it. Thank you

Stylll reviewed (approved):
Great job done. Please review and respond to my comments.

iverenshaguy committed:
feat(game-log): update game-log api endpoints

- update existing profile page endpoit data
- add get donations api endpoints
- update game history api endpoints
- use neccessary api endpoints in profile controller
- implement view for empty game history, leaderboard and donations
- implement cards for game history
- update tests

[Finishes #158487974]
